dev/set_matrix.py

```diff
@@ -340,11 +341,49 @@ def _requires_python(package: str, version: str) -> str | None:
     return None
 
 
-def infer_python_version(package: str, version: str) -> str:
+def _requires_python_from_repo(repo_url: str) -> str | None:
```

Comment:
Confirmed the right python version is inferred (3.11 for sklearn: https://github.com/scikit-learn/scikit-learn/pull/32646):

<img width="778" height="419" alt="image" src="https://github.com/user-attachments/assets/772dd62f-0e77-44d6-9c3c-0e6d54a15b19" />